### DISCUSSION TITLE
getting function signature mismatch while conveting obj to glb with -cc

### DISCUSSION THREAD

**Discussion by pramodwerea** [zeux/meshoptimizer#415] [General]
I am converting the obj file to GLB with flag -cc and for smaller sized obj file it seems fine. But when I tried to convert a big file which is around 900+MB I am getting the following runtime error. What could be the reason and how to handle it.

` RuntimeError: function signature mismatch
    at wasm-function[319]:48
    at wasm-function[196]:1003
    at wasm-function[217]:3874
    at wasm-function[219]:18
    at ../node_modules/gltfpack/library.js:55:33
    at process._tickCallback (internal/process/next_tick.js:68:7)
`

**Comment by zeux** [comment]
I'm not sure why the error is obscure, but my guess would be that this file is running into a 4GB Wasm heap limit. I'd recommend using native binaries to convert files, that you can download on release page (https://github.com/zeux/meshoptimizer/releases/tag/v0.17)

**Reply by pramodwerea** [reply]
Thanks, @zeux yes, it worked with the native binary.  But need to use within Nodejs, what should be the correct approach to use it?

**Reply by zeux** [reply]
You should be able to use `child_process` API for this: https://nodejs.org/api/child_process.html#child_processspawncommand-args-options
